bc_obps/reporting/service/report_new_entrant_service.py

```diff
@@ -0,0 +1,182 @@
+        emissions_data = ReportNewEntrantService._get_emissions_data(categories, report_new_entrant, naics_code)
+        result_data = (
+            model_to_dict(report_new_entrant, exclude=["selected_products"]) if report_new_entrant is not None else {}
+        )
+        return {"products": products_data, "emissions": emissions_data, "new_entrant_data": result_data}
+
+    @staticmethod
+    def _get_products_data(
+        regulated_products: List[RegulatedProductOut], report_new_entrant: Optional[ReportNewEntrant]  # Change here
+    ) -> List[Dict[str, Optional[str]]]:
+        """Helper to fetch product data"""
+        productions_map = (
+            {production.product.id: production.production_amount for production in report_new_entrant.productions.all()}
+            if report_new_entrant
+            else {}
+        )
+
+        return [
+            {
+                "id": str(product.id),
+                "name": product.name,
+                "unit": product.unit,
+                "production_amount": str(productions_map.get(product.id)) if productions_map.get(product.id) else None,
+            }
+            for product in regulated_products
+        ]
+
+    @staticmethod
+    def _get_emissions_data(
+        categories: List[EmissionCategory], report_new_entrant: Optional[ReportNewEntrant], naics_code: Optional[str]
+    ) -> list:
+        """Helper to fetch emissions data, categorized by type"""
+        emissions_map = (
+            {
+                emission.emission_category.id: emission.emission
+                for emission in report_new_entrant.report_new_entrant_emissions.all()
+            }
+            if report_new_entrant
+            else {}
+        )
+
+        emissions_by_type: dict = {
+            "basic": [],
+            "fuel_excluded": [],
+            "other_excluded": [],
+        }
+
+        for category in categories:
+            if (
+                naics_code == "324110"
+                and category.category_type == "other_excluded"
+                and "fat, oil and grease" in category.category_name.lower()
+            ):
+                continue
+
+            emissions_by_type[category.category_type].append(
+                {
+                    "id": category.id,
+                    "name": category.category_name,
+                    "emission": str(emissions_map.get(category.id)) if emissions_map.get(category.id) else None,
+                }
+            )
+
+        result = [
+            {
+                "name": "basic",
+                "title": "Emission categories after new entrant period began",
+                "emissionData": emissions_by_type["basic"],
+            },
+            {
+                "name": "fuel_excluded",
+                "title": "Emissions excluded by fuel type",
+                "emissionData": emissions_by_type["fuel_excluded"],
+            },
```

Comment:
These are frontend mapping things
What I would do, is give the frontend the data we have and the list of emission categories, and let it deal with that.
It'll make this code much simpler to write and test